definitions/mongodb.rb

```diff
@@ -203,24 +203,28 @@
 
   # replicaset
   if new_resource.is_replicaset && new_resource.auto_configure_replicaset
-    rs_nodes = search(
-      :node,
-      "mongodb_cluster_name:#{new_resource.replicaset['mongodb']['cluster_name']} AND \
-       mongodb_is_replicaset:true AND \
-       mongodb_shard_name:#{new_resource.replicaset['mongodb']['shard_name']} AND \
-       chef_environment:#{new_resource.replicaset.chef_environment}"
-    )
-
-    ruby_block 'config_replicaset' do
-      block do
-        MongoDB.configure_replicaset(new_resource.replicaset, replicaset_name, rs_nodes) unless new_resource.replicaset.nil?
+    if Chef::Config[:solo] and not chef_solo_search_installed?
+      Chef::Log.warn("This recipe uses search. Chef Solo does not support search unless you install the chef-solo-search cookbook.")
```

Comment:
Shouldn't this be a fatal error instead of just a log warn?
We were asked to auto configure a replica set, but we are unable to do so because chef solo & no chef-solo-search installed.
The rest will fail since we didn't fulfill our task: let's fail as soon as possible.